librz/magic/magic-test.c

```diff
@@ -0,0 +1,1372 @@
+			asctime_r(gmtime(&t), s);
+			break;
+		}
+		s[strcspn(s, "\n")] = '\0';
+		magic_add_result(ms, ml, "%s", s);
+		ms->offset += sizeof value;
+	}
+	return (result);
+}
+
+static int magic_test_type_uqdate(RzMagicLine *ml, RzMagicState *ms) {
+	uint64_t value;
+	int result;
+	time_t t;
+	char s[64];
+
+	if (magic_copy_from(ms, -1, &value, sizeof value) != 0)
+		return (0);
+	if (ml->type == MAGIC_TYPE_UBEQDATE ||
+		ml->type == MAGIC_TYPE_UBEQLDATE)
+		value = be64toh(value);
+	if (ml->type == MAGIC_TYPE_ULEQDATE ||
+		ml->type == MAGIC_TYPE_ULEQLDATE)
+		value = le64toh(value);
+
+	if (ml->type_operator == '&')
+		value &= (uint64_t)ml->type_operand;
+	else if (ml->type_operator == '-')
+		value -= (uint64_t)ml->type_operand;
+	else if (ml->type_operator == '+')
+		value += (uint64_t)ml->type_operand;
+	else if (ml->type_operator != ' ')
+		return (-1);
+
+	result = magic_test_unsigned(ml, value, (uint64_t)ml->test_unsigned);
+	if (result == !ml->test_not && ml->result != NULL) {
+		t = value;
+		switch (ml->type) {
+		case MAGIC_TYPE_UQLDATE:
+		case MAGIC_TYPE_ULEQLDATE:
+		case MAGIC_TYPE_UBEQLDATE:
+			ctime_r(&t, s);
+			break;
+		default:
+			asctime_r(gmtime(&t), s);
+			break;
+		}
+		s[strcspn(s, "\n")] = '\0';
+		magic_add_result(ms, ml, "%s", s);
+		ms->offset += sizeof value;
+	}
+	return (result);
+}
+
+static int magic_test_type_bestring16(__unused RzMagicLine *ml,
+	__unused RzMagicState *ms) {
+	return (-2);
+}
+
+static int magic_test_type_lestring16(__unused RzMagicLine *ml,
+	__unused RzMagicState *ms) {
+	return (-2);
+}
+
+static int magic_test_type_melong(__unused RzMagicLine *ml,
+	__unused RzMagicState *ms) {
+	return (-2);
+}
+
+static int magic_test_type_medate(__unused RzMagicLine *ml,
+	__unused RzMagicState *ms) {
+	return (-2);
+}
+
+static int magic_test_type_meldate(__unused RzMagicLine *ml,
+	__unused RzMagicState *ms) {
+	return (-2);
+}
+
+static int magic_test_type_regex(RzMagicLine *ml, RzMagicState *ms) {
+	const char *cp;
+	RzRegex *re;
+	int flags = 0, sflag = 0;
+
+	cp = &ml->type_string[(sizeof "regex") - 1];
+	if (*cp != '\0') {
+		if (*cp != '/')
+			return (-1);
+		cp++;
+		for (; *cp != '\0'; cp++) {
+			switch (*cp) {
+			case 's':
+				sflag = 1;
+				break;
+			case 'c':
+				flags |= RZ_REGEX_CASELESS;
+				break;
+			default:
+				return (-1);
+			}
+		}
+	}
+	re = rz_regex_new(ml->test_string, RZ_REGEX_EXTENDED | flags, 0, NULL);
+	if (!re) {
+		return -1;
+	}
+
+	bool res = true;
+	RzPVector *matches = rz_regex_match_first(re, ms->base, ms->size, ms->offset, RZ_REGEX_EXTENDED | flags);
+	if (!matches || rz_pvector_empty(matches)) {
+		res = false;
+	}
+	if (res == !ml->test_not) {
+		RzRegexMatch *match = NULL;
+		if (ml->result != NULL && res) {
+			match = rz_pvector_pop_front(matches);
+			magic_add_string(ms, ml, ms->base + match->start,
+				match->len - match->start);
+		}
+		if (res && match) {
+			if (sflag)
+				ms->offset = match->start;
+			else
+				ms->offset = match->len;
```

Comment:
```suggestion
				ms->offset = match->start + match->len;
```

`rm_eo` is the end offset. Not the `length` of the match.

Reply:
Thanks for reviewing that, thought it needs to be just replaced without a change. I was wrong.